workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/ProductCard.tsx

```diff
@@ -3,131 +3,221 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import ClayBadge from '@clayui/badge';
 import {useEffect, useState} from 'react';
 
+import './ProductCard.scss';
+
+import ClaySticker from '@clayui/sticker';
+
+import emptyPictureIcon from '../../../assets/icons/avatar.svg';
 import {getProductById} from '../../../utils/api';
+import {getCustomFieldValue} from '../../../utils/customFieldUtil';
+import {
+	getThumbnailByProductAttachment,
+	getValueFromSpecifications,
+} from '../../../utils/util';
+import {LicenseType} from '../enums/licenseType';
+import {Price} from '../enums/price';
+import {SkuOptions} from '../enums/skuOptions';
 
 interface ProductCardProps {
 	productId: number | null;
 	selectedAccount?: Account;
 	setProductToForm: (product: Product) => void;
-	showAccount: Boolean;
 }
 
 const ProductCard = ({
 	productId,
 	selectedAccount,
 	setProductToForm,
-	showAccount,
 }: ProductCardProps) => {
-	const [product, setProduct] = useState<Product[]>([]);
+	const [product, setProduct] = useState<Product>();
+	const [hasTrial, setHasTrial] = useState<boolean>(false);
+	const [basePrice, setBasePrice] = useState<Number | undefined>(undefined);
+
+	const productHasTrialSKU = (skus: SKU[]) => {
+		skus.map((sku) => {
+			const licenseUsageType = sku.skuOptions.find(
+				(option) =>
+					option.key.toLowerCase() === 'dxp-license-usage-type'
+			);
+			if (
+				licenseUsageType &&
+				licenseUsageType.value.toLowerCase() ===
+					SkuOptions.TRIAL.toLowerCase()
+			) {
+				setHasTrial(true);
+
+				return;
+			}
+
+			return;
+		});
```

Comment:
@ytuik since you are not returning anything from this map, you can replace to `forEach`.

The returns are not necessary also

Try that

```suggestion
		skus.forEach((sku) => {
			const licenseUsageType = sku.skuOptions.find(
				(option) =>
					option.key.toLowerCase() === 'dxp-license-usage-type'
			);
			
			if (
				licenseUsageType &&
				licenseUsageType.value.toLowerCase() ===
					SkuOptions.TRIAL.toLowerCase()
			) {
				setHasTrial(true);
			}
		});
```